### PR TITLE
Objdetect: QRCode pipeline: Added perf tests

### PR DESCRIPTION
**Pull request by allnes**
With opencv_extra PR [https://github.com/opencv/opencv_extra/pull/533](https://github.com/opencv/opencv_extra/pull/533)

### CODE REVIEW

File: modules/objdetect/perf/perf_qrcode_pipeline.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    SANITY_CHECK(corners);
+}
+
+#ifdef HAVE_QUIRC
+typedef ::perf::TestBaseWithParam< std::string > Perf_Objdetect_QRCode_Decode;
+PERF_TEST_P(Perf_Objdetect_QRCode_Decode,
+             qrcode_pipeline, ::testing::Values(
+               "version_1_down.jpg", "version_1_left.jpg", "version_1_right.jpg", "version_1_up.jpg", "version_1_top.jpg",
+               "version_5_down.jpg", "version_5_left.jpg", "version_5_right.jpg", "version_5_up.jpg", "version_5_top.jpg",
+               "russian.jpg", "kanji.jpg", "link_github_ocv.jpg", "link_ocv.jpg", "link_wiki_cv.jpg"
```
Comment (alalek) [inline modules/objdetect/perf/perf_qrcode_pipeline.cpp]: 1.

It is better to avoid copying of such list of images:
- use define
- or use single tests instantiation (see dnn/perf/perf_net.cpp):
```
PERF_TEST_P_(QRCode, detect)
{
...
}
PERF_TEST_P_(QRCode, decode)
{
...
}

INSTANTIATE_TEST_CASE_P(/*nothing*/, QRCode,
    ::testing::Values(
        ...
    )
);
```

2. Perhaps we should add performance tests for "negative" cases too (some images without qrcode (lena.png) + including random noise).

File: modules/objdetect/perf/perf_qrcode_pipeline.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
+#include "perf_precomp.hpp"
+
```
Comment (alalek) [inline modules/objdetect/perf/perf_qrcode_pipeline.cpp]:
```
// This file is part of OpenCV project.
// It is subject to the license terms in the LICENSE file found in the top-level directory
// of this distribution and at http://opencv.org/license.html.
```

**Comment by allnes** [comment]
@alalek please review me

File: modules/objdetect/perf/perf_qrcode_pipeline.cpp [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    std::string image_path = findDataFile(path_current_image);
+    Mat src = imread(image_path, IMREAD_GRAYSCALE), straight_barcode;
+    ASSERT_FALSE(src.empty()) << "Can't read image: " << image_path;
+
+    std::vector< Point > corners;
+    std::string decoded_info;
+    ASSERT_TRUE(detectQRCode(src, corners));
+
+    Mat not_qr_code = getNotQRCodeMat(GetParam());
```
Comment (alalek) [inline modules/objdetect/perf/perf_qrcode_pipeline.cpp]: It is better to fill corners in this generation function instead of using some image (pre-defined points should be fine).

Test inputs must be clear as much as possible and **don't expect correctness of other functions** (int this case - correctness of "detection" `detectQRCode()`).

Changes in other functions should not change the test result.

File: modules/objdetect/perf/perf_qrcode_pipeline.cpp [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    ::testing::Values(
+        "version_1_down.jpg", "version_1_left.jpg", "version_1_right.jpg", "version_1_up.jpg", "version_1_top.jpg",
+        "version_5_down.jpg", "version_5_left.jpg", "version_5_right.jpg", "version_5_up.jpg", "version_5_top.jpg",
+        "russian.jpg", "kanji.jpg", "link_github_ocv.jpg", "link_ocv.jpg", "link_wiki_cv.jpg"
+    )
+);
+
+Mat getNotQRCodeMat(std::string type_gen)
+{
+    Mat image(256, 256, CV_8UC1);
```
Comment (alalek) [inline modules/objdetect/perf/perf_qrcode_pipeline.cpp]: 256x256 is non standard camera resolution.
Try VGA / HD / FullHD instead.

**Comment by allnes** [comment]
**WIP**

**Comment by allnes** [comment]
@alalek please review me again
